src/mono/browser/runtime/pthreads/worker-thread.ts

```diff
@@ -210,14 +207,20 @@ export function replaceEmscriptenPThreadWorker(modulePThread: PThreadLibrary): v
     if (!WasmEnableThreads) return;
 
     const originalThreadInitTLS = modulePThread.threadInitTLS;
-    const original_emscripten_thread_init = (Module as any)["__emscripten_thread_init"];
 
-    (Module as any)["__emscripten_thread_init"] = (pthread_ptr: PThreadPtr, isMainBrowserThread: number, isMainRuntimeThread: number, canBlock: number) => {
-        on_emscripten_thread_init(pthread_ptr);
-        original_emscripten_thread_init(pthread_ptr, isMainBrowserThread, isMainRuntimeThread, canBlock);
-    };
     modulePThread.threadInitTLS = (): void => {
         originalThreadInitTLS();
         mono_wasm_pthread_on_pthread_created();
     };
-}
\ No newline at end of file
+}
+
+export function replaceEmscriptenPThreadWorker2(): void {
```

Comment:
This is an interesting name. Why we have `replaceEmscriptenPThreadWorker` and `replaceEmscriptenPThreadWorker2`? Let's figure out something more descriptive

Reply:
ideas welcome

Reply:
`replaceEmscriptenPThreadWorker` -> `replaceEmscriptenPThreadWorkerFromTLS` 
`replaceEmscriptenPThreadWorker2` -> `replaceEmscriptenPThreadWorkerFromModule`